### PR TITLE
Lesson 3

### PR DESCRIPTION
**Pull request by neu-milenko**

### CODE REVIEW

File: docs/lesson3.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+5. Inspect the code in the file. Notice that GitHub Copilot recognized test frameworks used in the solution.
+6. In the Chat View prompt, enter "how do I run these tests?".
+7. The response will include instructions to open the terminal in the test project folder and execute tests. Follow the instructions or executed the following (assuming current directory is **/eShop/):
+   ```bash
+   cd tests/Basket.UnitTests
+   dotnet test
+   ```
+
+**Expected Outcome:**  
+The existing and new unit tests should pass. If some test cases fail, you can use GitHub Copilot to fix the code or the failing unit tests.
```
Comment (dinmanchi) [inline docs/lesson3.md:80]: Can you elaborate on how to fix the code?

File: docs/lesson3.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+&emsp;- Create a new test project (change current directory to **tests/** before executing the commands)  
+&emsp;- Update the project file to include packages without specifying specific versions (remove coverlet.collector if referenced!)  
+&emsp;- Create a test class for the service inside the project folder  
+&emsp;- Run the tests
+   ```bash
+   dotnet test
+   ```
+
+**Expected Outcome:**  
+Newly created tests should pass. If some test cases fail, you can use GitHub Copilot to fix the code or the failing tests.
```
Comment (dinmanchi) [inline docs/lesson3.md:102]: same here. Just give an example of possible options

Comment (dinmanchi) [reply docs/lesson3.md:102]: Or refer to Prompting Techniques lesson. We can consolidated tips and tricks in that section as well

**Review by dinmanchi — APPROVED** [review]
Milenko. I have couple of minor comments. Please review and merge.

File: docs/lesson3.md (hunk 1 above)
Comment (neu-milenko) [reply docs/lesson3.md:80]: I can't predict the code or tests generated, so general advice there.

Comment (dinmanchi) [reply docs/lesson3.md:80]: I understand that. instead of saying fixing mention something around the lines of a additional prompting.

**Pull request merged**